src/gt4py/next/program_processors/runners/dace/utils.py

```diff
@@ -71,52 +71,68 @@ def is_connectivity_identifier(
 
 
 def is_connectivity_symbol(name: str, offset_provider_type: gtx_common.OffsetProviderType) -> bool:
-    m = SIZE_SYMBOL_RE.match(name) or STRIDE_SYMBOL_RE.match(name)
-    if m is None:
+    if (m_symbol := FIELD_SYMBOL_RE.match(name)) is None:
         return False
-    m = CONNECTIVITY_INDENTIFIER_RE.match(m[1])
-    if m is None:
+    if (m := CONNECTIVITY_INDENTIFIER_RE.match(m_symbol[1])) is None:
         return False
     return gtx_common.has_offset(offset_provider_type, m[1])
 
 
-def field_symbol_name(field_name: str, axis: int, sym: Literal["size", "stride"]) -> str:
-    return f"__{field_name}_{sym}_{axis}"
-
-
-def field_size_symbol_name(field_name: str, axis: int) -> str:
-    return field_symbol_name(field_name, axis, "size")
-
-
-def field_stride_symbol_name(field_name: str, axis: int) -> str:
-    return field_symbol_name(field_name, axis, "stride")
-
-
-def range_symbol(field_name: str, axis: str) -> str:
+def _field_symbol(
+    field_name: str,
+    dim: gtx_common.Dimension,
+    sym: Literal["size", "stride"],
+    offset_provider_type: Mapping[str, gtx_common.NeighborConnectivityType] | None,
+) -> dace.symbol:
+    if (m := CONNECTIVITY_INDENTIFIER_RE.match(field_name)) is None:
+        name = f"__{field_name}_{dim.value}_{sym}"
+    else:  # a connectivity field
+        assert offset_provider_type is not None
+        offset = m[1]
+        assert offset in offset_provider_type
```

Comment:
```suggestion
        assert m[1] in offset_provider_type
        offset = m[1]
```
Fully optional.
My only point is that now the asserts are consecutive.